### PR TITLE
Configuration reading/handling

### PR DESCRIPTION
**Pull request by nrsyed**
This PR removes the scriptconfig dependency found in the legacy sailon_client and introduces a default configuration that is updated with values from a dictionary or from a file in `ONDProtocol.__init__`. Because configuration handling will ultimately be the responsibility of tinker engine, the goal of this PR is only to provide a simple method for reading the existing configuration format without additional dependencies; it is not intended to be a robust means of configuration handling.

### CODE REVIEW

File: ond/ond.py [hunk 1]
```diff
@@ -95,3 +81,6 @@ def run_protocol(self):
                 test_params["image_classification_feedback"] = None
 
             # TODO: remainder of test functionality.
+
+if __name__ == "__main__":
+    protcol = ONDProtocol(None, None)
```
Comment (waxlamp) [inline ond/ond.py]:
```suggestion
    protocol = ONDProtocol(None, None)
```

**Review by waxlamp — CHANGES_REQUESTED** [review]
This seems like a bit of a work-in-progress, which is fine; by that metric this looks good to me. There's a minor typo that you may want to fix.

Since you're starting out with a static config here, a good next step might be simply to use the config mechanism of Tinker Engine instead of setting it up in-band this way. Let me know if that makes sense.

**Comment by nrsyed** [comment]
@waxlamp I've revamped this branch so that it imports the current iteration of tinker engine and leverages its config handling, like you suggested. This should simplify things considerably.

**Comment by waxlamp** [comment]
> @waxlamp I've revamped this branch so that it imports the current iteration of tinker engine and leverages its config handling, like you suggested. This should simplify things considerably.

Cool. Is this ready for a review then?

**Comment by nrsyed** [comment]
> > @waxlamp I've revamped this branch so that it imports the current iteration of tinker engine and leverages its config handling, like you suggested. This should simplify things considerably.
> 
> Cool. Is this ready for a review then?

Yep, it is.

**Review by waxlamp — APPROVED** [review]

**Pull request merged**